### DISCUSSION TITLE
Upload existing log files programmatically with C#

### DISCUSSION THREAD

**Discussion by mosmanski** [datalust/seq-tickets#1594] [Help]
Hello,

We are uploading our log files to Azure Blob Storage since our program runs on various networks, so we don't want to load directly to SEQ. We use Serilog, so they are structured properly already.

Is there a C# method using a client/api that can be used to upload existing log files to a SEQ server? I would like to avoid using the seqcli and was hoping there was a simpler way to upload log files with a console app or something.

Example of what I am hoping for:

var logFile = "C:\Temp\log1.log"
var seqApiClient = new SeqApiClient(SeqServerUrl);

seqApiClient.Ingest(File.ReadAllText(logFile));

Thanks, Mat

**Comment by nblumhardt** [comment]
Hi Mat,

Thanks for getting in touch! `seqcli` is the main way to do this, but you can alternatively load and send each line of the file using e.g. Serilog, or if you can format CLEF payloads, [send these directly to the HTTP API](https://docs.datalust.co/docs/posting-raw-events).

If you're dealing with a lot of data and complex log parsing, you might alternatively be able to derive something from the `seqcli` codebase.

Hope this helps,
Nick

**Comment by mosmanski** [comment]
Thanks Nick. I can code something using the sqlcli dotnet tool, I was just
hoping there was already a nuget package or something to handle this task
by passing in a file stream or something like that. Maybe in the future.

- Mat

**Comment by mosmanski** [comment]
Hi Nick,

Is the seqcli codebase in a public open source repository? If so, where?

Thanks, Mat

**Reply by nblumhardt** [reply]
Hi Mat - it's at https://github.com/datalust/seqcli 👍

**Comment by mosmanski** [comment]
Awesome, thanks!
